code/entities/chair/Armrest.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class Armrest : BaseNetworkable
+{
+    public enum States : int
+    {
+        Lowered,
+        Raising,
+        Raised,
+        Lowering
+    }
+
+    public enum Sides : int
+    {
+        Left,
+        Right
+    }
+
+    [Net]
+    public CinemaChair Chair { get; set; }
+    [Net]
+    public ModelEntity Cupholded { get; set; }
+    [Net]
+    public States State { get; set; } = States.Lowered;
+    [Net]
+    public Sides Side { get; set; }
+
+    public string CupholderBoneName => Side switch
+    {
+        Sides.Left  => "cupholder_l",
+        _           => "cupholder_r"
+    };
+
+    public string ArmrestToggleParameter => Side switch
+    {
+        Sides.Left => "toggle_left_armrest",
+        _ => "toggle_right_armrest"
+    };
+
+    public void HandleAnimTag(string tag)
+    {
+        if (tag.Contains("launch"))
+        {
+            Launch();
+            State = States.Raised;
+        }
+        else if (tag.Contains("lowered"))
+        {
+            State = States.Lowered;
+        }
+    }
+
+    public void Toggle()
+    {
+        var originalValue = Chair.GetAnimParameterBool(ArmrestToggleParameter);
+        Chair.SetAnimParameter(ArmrestToggleParameter, !originalValue);
+        // If toggle_seat was true but now is false, we are now lowering the seat.
+        State = originalValue
+            ? States.Lowering
+            : States.Raising;
+    }
+
+    public void CupholdEntity(ModelEntity entity)
+    {
+        if (State != States.Lowered)
+        {
+            return;
+        }
+        Cupholded = entity;
```

Comment:
What happens if there is already an entity in the cupholder? I know right now you handle this with the user (caller) code, but we could handle that here too to be extra safe.

Reply:
Made a change so that an already held entity will be launched. This uses a different launch direction and impulse magnitude, so I made LaunchHeldEntity use parameters for this.

https://github.com/tech-nawar/sbox-cinema/assets/8889285/1b098263-b028-435c-a8ab-aa78c03c53a1